### DISCUSSION TITLE
Pooler won't start in docker self-hosted

### DISCUSSION THREAD

**Discussion by AbhiSawantB6** [prisma/prisma#25473] [Q&A] [answered]
### Question

I am trying to run supabase in docker.  I found the instructions here https://supabase.com/docs/guides/self-hosting/docker
My git log shows commit 9f9b083ba5a31112ec8ceeedeb43ea6fa5958996.

When I try to start the downloaded repo. I see an issue with the pooler startup and subsequently pgadmin won't connect to 5432 and complains that user postgres is missing.

This is what I see in the pooler startup log.  I tried to update the pooler.exs, but that doesn't help.

: unexpected token: carriage return (column 4, code point U+000D)
    |
 30 | end
    |    ^
    (elixir 1.14.3) lib/code.ex:424: Code.validated_eval_string/3
** (SyntaxError) nofile:30:4: unexpected token: carriage return (column 4, code point U+000D)
    |
 30 | end
    |    ^
    (elixir 1.14.3) lib/code.ex:424: Code.validated_eval_string/3
** (SyntaxError) nofile:30:4: unexpected token: carriage return (column 4, code point U+000D)
    |
 30 | end
    |    ^
    (elixir 1.14.3) lib/code.ex:424: Code.validated_eval_string/3
** (SyntaxError) nofile:30:4: unexpected token: carriage return (column 4, code point U+000D)
    |
 30 | end
    |    ^
    (elixir 1.14.3) lib/code.ex:424: Code.validated_eval_string/3
** (SyntaxError) nofile:30:4: unexpected token: carriage return (column 4, code point U+000D)
    |
 30 | end
    |    ^
    (elixir 1.14.3) lib/code.ex:424: Code.validated_eval_string/3

### How to reproduce (optional)

Using Docker Desktop 4.34.3 (170107) is currently the newest version available.  Run up the github instance as documented.

# Get the code
git clone --depth 1 https://github.com/supabase/supabase

# Go to the docker folder
cd supabase/docker

# Copy the fake env vars
cp .env.example .env

# Pull the latest images
docker compose pull

# Start the services (in detached mode)
docker compose up -d

View supabase pooler logs.  It doesn't start.  

Try to update <root>\supabase\docker\volumes\pooler\pooler.exs and add a new line at the end.

Does not help.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @AbhiSawantB6 👋 

Are you running the example on Windows Operating System? If yes, Open the `pooler.exs` file in Visual Studio Code, and check the bottom-right corner of the editor. If it shows `CRLF`, click on it and select `LF` to change line endings. After fixing the line endings, try restarting your Docker containers:

```
# Stop the containers
docker compose down

# Start the services again
docker compose up -d
```
You can inspect the logs using `docker compose logs` to see if the services start without error.

**Reply by AbhiSawantB6** [reply]
Yes, I am on Windows.  This worked thank you!
